diagnostic.md

```diff
@@ -5,74 +5,91 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What is the purpose of a backend?
 
 ```bash
-// your response here
+A back end is used to store dataa and in game example, game states and views.
 ```
 
 Which layer in the MVC pattern is used by the controller to fetch data?
 
 ```bash
-// your response here
+model
 ```
 
 Which layer in the MVC pattern communicates with the model?
 
 ```bash
-// your response here
+controller
 ```
 
 Why don't we use views in our interpretation of the MVC pattern?
 
 ```bash
-// your response here
+
 ```
 
 What does C.R.U.D stand for?
 
 ```bash
-// your response here
+C - create
+R - read?
+U - Update
+D - Delete
 ```
 
 In which part of the MVC pattern can we find C.R.U.D actions?
 
 ```bash
-// your response here
+controller
 ```
 
 List at least 5 standard rails actions that C.R.U.D requests correspond to?
 
 ```bash
-// your response here
+index
+show
+create
+update
+destroy
 ```
 
 A user action fires a `GET` request for `/people/1`. Explain in detail each step
 required for data to be returned to the client. (bullet points or ordered list)
 
 ```bash
-// your response here
+1. A user opens his browser, types in a URL.
+2. When the user presses Enter, the browser sends a GET request for that URL.
+3. The GET request hits the Rails router. The router maps the URL to the
+   correct controller action to handle the request.
+4. The action receives the GET request and passes it on to the view.
+5. The view renders the page as HTML.
+6. The controller sends the HTML back to the browser.
+   The page loads and the user sees the page with the form.
 ```
 
 What is the command to generate a new rails-api app?
 
 ```bash
-// your response here
+rails-api new blog_app --skip-javascript --skip-sprockets --skip-turbolinks
+--skip-test-unit --database=postgresql
 ```
```

Comment:
That's awesome, assuming you know exactly what each of those commands do and why you're using them. If not, being that verbose isn't necessary.

simply `bundle exec rails-api new appname` or `bundle exec rails new appname --api` is all you need.

the `rails-api` and/or `--api` flags are just telling rails we're creating a strictly api project, and won't generate any of the view associated files that we don't need. 

Make sense?